### ISSUE TITLE
Use Asynchronous Calls instead of sync when doing Network calls

### ISSUE THREAD

**Issue by srini1978** [Green-Software-Foundation/green-software-patterns#76]
**Describe the pattern you'd like to propose**
A clear and concise description of what you think the pattern should be.

When making calls across process boundaries to either databases or file systems or REST APIs, use asynchronous calling patterns compared to synchronous calls. In C# this feature is available either through the async/await syntax or through Task Parallel Library

**Describe specific emission impact from this pattern**
A clear and concise description of how you see this pattern impacting software emissions.
Using asynchronous patterns frees the calling thread from being blocked on the response and hence additional work is achieved without CPU cycles being consumed. Hence this pattern drives optimal utilization of the CPU and Memory cycles contributing to lesser Energy or E value.  Reducing the E value reduces the SCI score, 

**References to this pattern**
Is this a new concept or does this pattern exist somewhere already? Please provide any references you have to existing examples.
New concept

**Additional context**
Add any other context or screenshots about the proposed pattern here.

**Comment by dubrie** [comment]
Thank you @srini1978 for opening this issue. I feel this would be a good pattern for the Web category and would have positive impact on energy efficiency. Can you create a new pattern following the updated Template at https://github.com/Green-Software-Foundation/green-software-patterns/blob/dev/TEMPLATE.md?

**Comment by srini1978** [comment]
@dubrie  Created this PR -   https://github.com/Green-Software-Foundation/green-software-patterns/pull/130 however I think it is showing 199 commits which I know is not right. Can u help?

**Comment by srini1978** [comment]
@dubrie  I am unable to fork and create a PR. Is that because of access permissions ? Should i only be opening a issue ?

**Comment by markus-seidl** [comment]
@srini1978 I've just checked and could fork the repo without any issues on my private account - no permissions needed.

**Comment by srini1978** [comment]
@markus-seidl  How do I add the changes I have made into my dev branch ? 

https://github.com/Green-Software-Foundation/green-software-patterns-dev/blob/3082514aefa56b1688775e603bdc0532320f390f/docs/catalog/web/Use%20async%20network%20calls.md

**Comment by markus-seidl** [comment]
It seems that you have write rights in this repository/organization - @dubrie @jawache Intended?

If you compare the branches like this : https://github.com/Green-Software-Foundation/green-software-patterns/compare/dev...Green-Software-Foundation:green-software-patterns-dev:dev you'll get a button to "Create Pull Request". Make sure that you select the original green-software-patterns repository as base and your branch as source (left - right). You can see below the files that you are going to send with the PR, make sure that is the correct content.

**Comment by dubrie** [comment]
Hey @srini1978, I created a branch off of this issue for you that is based on the dev branch for the patterns repo. Are you able to run these commands from a cloned green-software-patterns repo?

`git fetch origin`   
`git checkout 76b-use-asynchronous-calls-instead-of-sync-when-doing-network-calls`

**Comment by markus-ntt-seidl** [comment]
I'm using the pattern markdown linked above and creating a new PR
